rajawali/src/main/java/c/org/rajawali3d/engine/EngineImpl.java

```diff
@@ -220,105 +244,104 @@ public void onRenderSurfaceCreated(EGLConfig config, GL10 gl, int width, int hei
+                renderModel.onFrameStart(deltaTime);
+            }
+            for (RenderView renderView : renderViews) {
+                renderView.onFrameStart(deltaTime);
+            }
+        } catch (InterruptedException e) {
+            Log.e(TAG, "startFrame(): Frame skipped due to thread interruption.");
+        }
     }
 
-    /**
-     * Called by {@link #onRenderFrame(GL10)} to render the next frame. This is called prior to the current scene's
-     * {@link Scene#render(long, double)} method.
-     *
-     * @param ellapsedRealtime {@code long} The total ellapsed rendering time in milliseconds.
-     * @param deltaTime        {@code double} The time passes since the last frame, in seconds.
-     */
-    protected void onRender(final long ellapsedRealtime, final double deltaTime) throws InterruptedException {
-        render(ellapsedRealtime, deltaTime);
+    protected void renderViews() {
+        try {
+            for (RenderView renderView : renderViews) {
+                renderView.onRenderView();
+            }
+        } catch (InterruptedException e) {
+            Log.e(TAG, "renderViews(): Frame skipped due to thread interruption.");
+        }
     }
 
-    /**
-     * Called by {@link #onRender(long, double)} to render the next frame.
-     *
-     * @param ellapsedRealtime {@code long} Render ellapsed time in milliseconds.
-     * @param deltaTime        {@code double} Time passed since last frame, in seconds.
-     */
-    protected void render(final long ellapsedRealtime, final double deltaTime) throws InterruptedException {
-        if (currentRenderable != null) {
-            currentRenderable.render(ellapsedRealtime, deltaTime);
-        } else {
-            Log.w(TAG, "No renderable has been set!");
+    protected void endFrame(double deltaTime) {
+        try {
+            // TODO not sure if order matters here, just swapping from startFrame() on principle...
+            for (RenderView renderView : renderViews) {
+                renderView.onFrameEnd(deltaTime);
+            }
+            for (RenderModel renderModel : renderModels) {
+                renderModel.onFrameEnd(deltaTime);
+            }
+        } catch (InterruptedException e) {
+            Log.e(TAG, "endFrame(): Frame skipped due to thread interruption.");
         }
     }
 
-    /**
-     * Switches which {@link Renderable} is to be rendered.
-     *
-     * @param nextRenderable The {@link Renderable} to switch to.
-     */
-    @GuardedBy("nextRenderableLock")
-    @GLThread
-    void switchRenderable(@NonNull Renderable nextRenderable) {
-        RajLog.d("Switching from renderable: " + currentRenderable + " to renderable: " + nextRenderable);
-        currentRenderable = nextRenderable;
-        currentRenderable.restoreForNewContextIfNeeded();
+    @Override
+    public void onOffsetsChanged(float xOffset, float yOffset, float xOffsetStep, float yOffsetStep, int xPixelOffset, int yPixelOffset) {
```

Comment:
We need to create an interface and add a `@Nullable` delegate member to this class to propagate this and `onTouchEvent(MotionEvent)` calls to optional user code since we are removing the extension of what used to be `Renderer` classes.